src/modules/0.0.22/aws_s3/mappers/bucket.ts

```diff
@@ -38,74 +75,131 @@ export class BucketMapper extends MapperBase<Bucket> {
+          if (!location) location = 'us-east-1';
+          if (enabledRegions.includes(location)) {
+            // read policy
+            const input: GetBucketPolicyCommandInput = {
+              Bucket: rawBucket.Name,
+            };
 
-          if (bucketPolicy && bucketPolicy.Policy) {
-            b.policyDocument = JSON.parse(bucketPolicy.Policy);
-          } else {
-            b.policyDocument = undefined;
+            const bucketPolicy = await this.getBucketPolicy(client.s3Client, input);
+            const b: Bucket = this.module.bucket.bucketMapper(rawBucket, location);
+
+            if (bucketPolicy && bucketPolicy.Policy) {
+              b.policyDocument = JSON.parse(bucketPolicy.Policy);
+            } else {
+              b.policyDocument = undefined;
+            }
+            out.push(b);
           }
-          out.push(b);
         }
       }
       return out;
     },
     updateOrReplace: (a: Bucket, b: Bucket) => {
-      if (!Object.is(a.policyDocument, b.policyDocument)) return 'update';
+      if (!policiesAreSame(a.policyDocument ?? [], b.policyDocument ?? [])) return 'update';
       else return 'replace';
     },
     // TODO: With the model this simple it is actually impossible to really update this thing
     // as the user can't change the create timestamp themselves, so just restore the record from
     // the cloud cache and force it into the DB cache
     update: async (es: Bucket[], ctx: Context) => {
-      const client = (await ctx.getAwsClient()) as AWS;
       const out = [];
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
+
       for (const e of es) {
-        const cloudRecord = ctx?.memo?.cloud?.Bucket?.[e.name ?? ''];
-        const isUpdate = Object.is(this.module.bucket.cloud.updateOrReplace(cloudRecord, e), 'update');
-        if (isUpdate) {
-          e.createdAt = cloudRecord.createdAt;
-          e.policyDocument = await this.module.bucket.createBucketPolicy(client.s3Client, e, ctx);
-          out.push(e);
-        } else {
-          // Replace if name has changed
-          await this.module.bucket.db.update(cloudRecord, ctx);
-          ctx.memo.db.Bucket[cloudRecord.name] = cloudRecord;
-          out.push(cloudRecord);
+        if (enabledRegions.includes(e.region)) {
+          const client = (await ctx.getAwsClient(e.region)) as AWS;
+          const cloudRecord = ctx?.memo?.cloud?.Bucket?.[this.entityId(e)];
+          const isUpdate = Object.is(this.module.bucket.cloud.updateOrReplace(cloudRecord, e), 'update');
+          if (isUpdate) {
+            e.createdAt = cloudRecord.createdAt;
+            e.policyDocument = await this.module.bucket.createBucketPolicy(client.s3Client, e, ctx);
+            out.push(e);
+          } else {
+            // we cannot modify bucket name or region of the bucket, replace it
+            await this.module.bucket.db.update(cloudRecord, ctx);
+            ctx.memo.db.Bucket[cloudRecord.name] = cloudRecord;
```

Comment:
We should always treat the memo with the same entityId! Maybe this should be something like
```suggestion
            ctx.memo.db.Bucket[this.entityId(cloudRecord)] = cloudRecord;
```